### PR TITLE
Human Npc Improvements

### PR DESCRIPTION
**Pull request by alek-kam-robotec-ai**
This pull request:
  - extends the current Human Npc with the repeat on traversed toggle
  - adds a way to create random waypoint paths with the WaypointSelectorComponent

### CODE REVIEW

File: Project/Gem/Source/HumanNpc/NpcNavigatorBus.h [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+        //////////////////////////////////////////////////////////////////////////
+        // EBusTraits overrides
+        static constexpr AZ::EBusHandlerPolicy HandlerPolicy = AZ::EBusHandlerPolicy::Single;
+        static constexpr AZ::EBusAddressPolicy AddressPolicy = AZ::EBusAddressPolicy::ById;
+        using BusIdType = AZ::EntityId;
+        //////////////////////////////////////////////////////////////////////////
+    };
+
+    using NpcNavigatorRequestBus = AZ::EBus<NpcNavigatorRequests, NpcNavigatorRequestBusTraits>;
+} // namespace ROS2::Demo
```
Comment (michalpelka) [inline Project/Gem/Source/HumanNpc/NpcNavigatorBus.h:30]: Nit : new line.

File: Project/Gem/Source/HumanNpc/WaypointSelectorComponent.h [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+        void Deactivate() override;
+
+    private:
+        // AZ::EntityBus overrides
+        void OnEntityActivated(const AZ::EntityId&) override;
+
+        AZStd::vector<AZ::EntityId> SelectWaypointPah();
+
+        size_t m_seed{ 0 };
+        std::mt19937 m_mersenneTwister{ m_seed };
```
Comment (michalpelka) [inline Project/Gem/Source/HumanNpc/WaypointSelectorComponent.h:40]: Consider using random generator from AZStd.

Comment (alek-kam-robotec-ai) [reply Project/Gem/Source/HumanNpc/WaypointSelectorComponent.h:40]: The mersenne twister does not appear to be implemented in the AZStd namespace

File: Project/Gem/Source/HumanNpc/NpcNavigatorBus.h [hunk 3]
```diff
@@ -0,0 +1,30 @@
+#pragma once
```
Comment (aniafafe) [inline Project/Gem/Source/HumanNpc/NpcNavigatorBus.h:1]: Missing licence.

File: Project/Gem/Source/HumanNpc/NpcNavigatorComponent.cpp [hunk 4]
```diff
@@ -20,6 +20,7 @@ namespace ROS2::Demo
             serialize->Class<NpcNavigatorComponent, AZ::Component>()
                 ->Version(1)
                 ->Field("Debug Mode", &NpcNavigatorComponent::m_debugMode)
+                ->Field("Restart on traversed", &NpcNavigatorComponent::m_restartOnTraversed)
```
Comment (aniafafe) [inline Project/Gem/Source/HumanNpc/NpcNavigatorComponent.cpp:23]: Missing licence.

File: Project/Gem/Source/HumanNpc/NpcNavigatorComponent.h [hunk 5]
```diff
@@ -1,6 +1,5 @@
 #pragma once
 
-#include "WaypointComponent.h"
```
Comment (aniafafe) [inline Project/Gem/Source/HumanNpc/NpcNavigatorComponent.h:3]: Missing license.

File: Project/Gem/Source/HumanNpc/WaypointSelectorComponent.h [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+        AZStd::vector<AZ::EntityId> m_humanNpcs, m_waypointEntities;
```
Comment (aniafafe) [inline Project/Gem/Source/HumanNpc/WaypointSelectorComponent.h:41]: There is no need to add entities to the name of m_waypoints. The type makes it clear enough.

File: Project/Gem/Source/ROSCon2023DemoModule.cpp [hunk 7]
```diff
@@ -14,6 +14,8 @@
 #include <HumanNpc/NpcNavigatorComponent.h>
 #include <HumanNpc/WaypointComponent.h>
 #include <Scripting/FoilWrapperController.h>
+#include <HumanNpc/WaypointSelectorComponent.h>
```
Comment (aniafafe) [inline Project/Gem/Source/ROSCon2023DemoModule.cpp:17]: Missing license.

File: Project/Gem/Source/HumanNpc/WaypointSelectorComponent.h [hunk 8]
```diff
@@ -0,0 +1,43 @@
+#pragma once
```
Comment (aniafafe) [inline Project/Gem/Source/HumanNpc/WaypointSelectorComponent.h:1]: Missing license.

File: Project/Gem/Source/HumanNpc/WaypointSelectorComponent.cpp [hunk 9]
```diff
@@ -0,0 +1,91 @@
+#include <HumanNpc/NpcNavigatorBus.h>
```
Comment (aniafafe) [inline Project/Gem/Source/HumanNpc/WaypointSelectorComponent.cpp:1]: Missing license

File: Project/Gem/Source/HumanNpc/WaypointComponent.h [hunk 10]
```diff
@@ -15,6 +15,9 @@ namespace ROS2::Demo
         float m_idleTime{ 0 };
     };
 
+    //! Component that signifies a waypoint.
```
Comment (aniafafe) [inline Project/Gem/Source/HumanNpc/WaypointComponent.h:18]: Missing license

**Review by aniafafe — APPROVED** [review]
LGTM, just please add the license.

**Comment by alek-kam-robotec-ai** [comment]
@michalpelka I've applied review changes and rebased the branch.

**Review by michalpelka — APPROVED** [review]

**Pull request merged**